pydatastructs/utils/misc_util.py

```diff
@@ -40,3 +40,30 @@ def __str__(self):
         Used for printing.
         """
         return str((self.left, self.key, self.data, self.right))
+
+
+class GraphNode(object):
+    """
+    Represents node in Graphs.
+
+    Parameters
+    ==========
+
+    data
+        Any valid data to be stored in the node.
+    key
+        Required for comparison operations.
+    """
+
+    __slots__ = ['key', 'data']
+
+    def __new__(cls, key, data):
```

Comment:
I think allowing adjacency list in the constructor of `GraphNode` would be good. Something like,
```python
def __new__(cls, key, data, adj_list):
   ...
```

Reply:
Agreed on this.